packages/camera/android/src/test/java/io/flutter/plugins/camera/FilePathValidatorTest.java

```diff
@@ -0,0 +1,97 @@
+package io.flutter.plugins.camera;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.*;
+
+import java.io.File;
+import org.junit.Test;
+
+public class FilePathValidatorTest {
+
+  @Test
+  public void validate_ReturnsIsDirectory() {
+    FilePathValidator.FileFactory mockFactory = mock(FilePathValidator.FileFactory.class);
+    File mockFile = mock(File.class);
+    String filePath = "foo";
+
+    when(mockFile.exists()).thenReturn(true);
+    when(mockFile.isDirectory()).thenReturn(true);
+    when(mockFactory.makeFile(filePath)).thenReturn(mockFile);
+
+    FilePathValidator validator = new FilePathValidator(filePath, mockFactory);
+    validator.validate();
+
+    assertEquals(
+        "filePathInvalid",
+        "File at path '"
+            + filePath
+            + "' is a directory. This path should be a destination file path.",
+        validator.getErrorMessage());
+  }
+
+  @Test
+  public void validate_ReturnsFileAlreadyExists() {
+    FilePathValidator.FileFactory mockFactory = mock(FilePathValidator.FileFactory.class);
+    File mockFile = mock(File.class);
+    String filePath = "foo";
+
+    when(mockFile.exists()).thenReturn(true);
+    when(mockFactory.makeFile(filePath)).thenReturn(mockFile);
+
+    FilePathValidator validator = new FilePathValidator(filePath, mockFactory);
+    validator.validate();
+
+    assertEquals(
+        "File at path '" + filePath + "' already exists. Cannot overwrite.",
```

Comment:
nit:
Since "foo" is already known to be the file path, we typically just add it to the String in tests
```suggestion
        "File at path 'foo' already exists. Cannot overwrite.",
```